### PR TITLE
convert recipe to noarch

### PR DESCRIPTION
**Pull request by ngam**
Closes #96 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by weiji14** [comment]
Could we just make this package `noarch`? I tried running `grayskull pypi parcels` and it gave a noarch package by default, though not sure if it's ok.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,36 +10,29 @@ source:
...
-    - python
+    - python  >=3.8
     - cftime
     - cgen
     - dask
-    - enum34  # [py2k]
-    - gcc_linux-64  # [linux]
-    - clang_osx-64  # [osx]
-    - m2w64-toolchain  # [win]
+    - enum34
```
Comment (weiji14) [inline recipe/meta.yaml:32]: Can remove [`enum34`](https://pypi.org/project/enum34/), this is for Python<=3.3 only.
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,36 +10,28 @@ source:
...
   run:
-    - python
+    - python  >=3.8
     - cftime
     - cgen
     - dask
-    - enum34  # [py2k]
-    - gcc_linux-64  # [linux]
-    - clang_osx-64  # [osx]
-    - m2w64-toolchain  # [win]
```
Comment (weiji14) [inline recipe/meta.yaml:38]: It's possible to have os-specific dependencies with noarch, see https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies.

**Comment by weiji14** [comment]
> I don't the noarch will work; they do have some c/cpp code going on: https://github.com/OceanParcels/parcels/tree/master/parcels/include

Hmm yeah, I noticed that too, but it was worth a shot :laughing: This recipe is pulling the tar.gz from GitHub, so we should be able to run the unit tests to check.

**Comment by ocefpaf** [comment]
The package requires compilers at runtime. They are, in a way, optional, so I wonder if we could do a multiple output approach here, a core/base without the compilers and a complete one.

**Review by varlackc — COMMENTED** [review]
There seem to be some redirects in the URLs in the `meta.yaml` file.

  The home URL should probably use `https` instead of `http`

  https://github.com/conda-forge/parcels-feedstock/blob/bee3646c76b1f2a98bc876418939261850643943/recipe/meta.yaml#L60

  Probably we should use the following URL instead:

  home: https://github.com/OceanParcels/parcels

  In addition, we probably should modify as well the `doc_url`

  https://github.com/conda-forge/parcels-feedstock/blob/bee3646c76b1f2a98bc876418939261850643943/recipe/meta.yaml#L72

  Probably we should use the following URL instead:

  doc_url: https://oceanparcels.org/

**Review by erikvansebille — APPROVED** [review]
This looks good to me, thanks for all this effort! However, I would also like to see @willirath weigh in, who was the original creator of this condo recipe.

**Comment by erikvansebille** [comment]
Actually, one question: I do see a check for Windows (`parcels-feedstock (win win_64_)`) and for linux (`parcels-feedstock (linux linux_64_)`), but not for macOS. Is that intentional? Should we not also check macOS?

**Comment by ocefpaf** [comment]
> @ocefpaf we could do without outputs.

You are correct, with the modern "semi-arch" we can achieve better results.

**Comment by willirath** [comment]
Thanks for pinging me. I like the semi-arch idea, but I'm a little sceptical if we really cover the implications of the [new way to depend on the compilers](https://github.com/conda-forge/parcels-feedstock/pull/98/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR34-R36)
with the current tests.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -50,14 +48,16 @@ requirements:
     - zarr >=2.11.0
 
 test:
+  requires:
+    - pip
   imports:
     - parcels
   commands:
     - curl wget https://raw.githubusercontent.com/OceanParcels/parcels/master/docs/examples/example_peninsula.py > example_peninsula.py
     - python example_peninsula.py
```
[the remark this answers is not in the document]
Comment (willirath) [reply recipe/meta.yaml:57]: (Note this also drops an erroneous `wget` from the `curl` call in passing.

**Comment by erikvansebille** [comment]
> > As already mentioned in the comments, I'd very much like to explicitly test that compilers are present and found on OSX as well.
> 
> Let me know if the edits are sufficient. Thanks!

Thanks for also adding the osx test, @ngam. Glad to see that the `python example_peninsula.py` call works with the built-in compilers. 

I think this should be fine now to merge this PR; do you agree @willirath?

**Review by varlackc — APPROVED** [review]
Looks Good To Me!

**Comment by willirath** [comment]
Looks good to me as well

**Comment by willirath** [comment]
No idea how I can formally approve on mobile. Feel free to ignore my review and merge.

**Pull request merged**